mne/beamformer/_lcmv.py

```diff
@@ -24,6 +24,16 @@
 from ..channels.channels import _contains_ch_type
 
 
+def _deprecate_picks(data_obj, picks):
+    if picks is not None:
+        warn('Specifying picks is deprecated and will be removed in 0.16. '
+             'Specifying the selection of channels in info and setting picks '
+             'to None will remove this warning.',
+             DeprecationWarning)
+
+        data_obj.info = pick_info(data_obj.info, picks)
```

Comment:
I still don't like modifying only the `info` part of the object -- it's not very clean and will be prone to breakage / misunderstanding by the next dev. (It is reasonable to assume no such inconsistencies exist when looking at a given Evoked/Epochs/Raw object.)

Instead of doing that, can you return the modified info object instead and use it?